### DISCUSSION TITLE
Handling scheduled jobs for non-existing IKE SA

### DISCUSSION THREAD

**Discussion by fasilz** [strongswan/strongswan#1788] [General]
I have read this comment https://github.com/strongswan/strongswan/issues/853#issuecomment-1058921280, and would like to learn if there is any suggested way to dequeue the jobs and cancel them?  I have looked at the cancel() in the processor but looks like it's designed for shutdown. May be 	lib->scheduler->flush(lib->scheduler)?

**Comment by tobiasbrunner** [comment]
There is currently no way to cancel or dequeue jobs. They just have to be implemented in a way that they are basically no-ops if they run later but aren't required anymore (e.g. because the related IKE_SA has already been terminated).
